piccolo_admin/endpoints.py

```diff
@@ -709,139 +809,93 @@ def __init__(
                     ),
-                )
+                    provider=InMemoryLimitProvider(limit=20, timespan=300),
+                    # Note: this provider is hard-coded; we may consider using self._rate_limit_provider instead
+                ),
+            )
 
-        #######################################################################
+    def _init_index_template(self):
+        with open(os.path.join(ASSET_PATH, "index.html")) as f:
+            return f.read()
 
-        public_app = FastAPI(
-            redoc_url=None,
-            docs_url=None,
-            debug=debug,
-            exception_handlers={500: log_error},
-        )
-        public_app.mount("/docs/", swagger_ui(schema_url="../openapi.json"))
+    def _check_media_storage(self):
+        # Make sure media storage is configured properly.
 
-        if not rate_limit_provider:
-            rate_limit_provider = InMemoryLimitProvider(
-                limit=20,
-                timespan=300,
+        media_storage = [
+            i
+            for i in itertools.chain(
+                *[
+                    table_config.media_storage or []
+                    for table_config in self.table_configs
+                ]
             )
+        ]
 
-        public_app.mount(
-            path="/login/",
-            # This rate limiting is to prevent brute forcing password login,
-            # and MFA codes.
-            app=RateLimitingMiddleware(
-                app=session_login(
-                    auth_table=self.auth_table,
-                    session_table=session_table,
-                    session_expiry=session_expiry,
-                    max_session_expiry=max_session_expiry,
-                    redirect_to=None,
-                    production=production,
-                    mfa_providers=mfa_providers,
-                ),
-                provider=rate_limit_provider,
-            ),
-        )
-
-        public_app.add_route(
-            path="/logout/",
-            route=session_logout(session_table=session_table),  # type: ignore
-            methods=["POST"],
-        )
-
-        # We make the meta endpoint available without auth, because it contains
-        # the site name.
-        public_app.add_api_route(
-            "/meta/", endpoint=self.get_meta, tags=["Meta"]  # type: ignore
-        )
+        if len(media_storage) != len(set(media_storage)):
+            raise ValueError(
+                "Media storage is misconfigured - multiple columns are saving "
+                "to the same location."
+            )
 
-        # The translations are public, because we need them on the login page.
-        public_app.add_api_route(
-            "/translations/",
-            endpoint=self.get_translation_list,  # type: ignore
-            methods=["GET"],
-            tags=["Translations"],
-            response_model=TranslationListResponse,
-        )
+    def _init_table_configs(self, tables) -> list[TableConfig]:
```

Comment:
```suggestion
    def _init_table_configs(self, tables: t.Sequence[t.Union[t.Type[Table], TableConfig]]) -> t.List[TableConfig]:
```
Also as we support `python 3.8` still we need to use the typing module types